disnake/state.py

```diff
@@ -782,14 +783,22 @@ def parse_message_create(self, data: gateway.MessageCreateEvent) -> None:
         if self._messages is not None:
             self._messages.append(message)
         # we ensure that the channel is a type that implements last_message_id
-        if channel and channel.__class__ in (TextChannel, Thread, VoiceChannel):
-            channel.last_message_id = message.id  # type: ignore
+        if channel:
+            if channel.__class__ in (TextChannel, Thread, VoiceChannel):
+                channel.last_message_id = message.id  # type: ignore
+            if channel.__class__ is Thread and message.type is not MessageType.thread_starter_message:
+                channel.total_message_sent += 1
+                channel.message_count += 1
```

Comment:
@onerandomusername using `type()` in both ifs conditions we could remove the `# type: ignore`

Reply:
The difference between `x.__class__` and `type(x)` is close to none at that scale (difference of a few nanoseconds), so let's go with `type(x)` here as pyright can properly narrow the type that way.